vecnet/vecnet_test.go

```diff
@@ -46,3 +48,77 @@ func TestReadFromSanity(t *testing.T) {
 		t.Errorf("ReadFrom() = (%v, %#v), want (%v, %#v)", s1, s2, s[:10], s[10:])
 	}
 }
+
+func TestReadFromNetwork(t *testing.T) {
+	testCloser := func(closer io.Closer) {
+		t.Helper()
+		if err := closer.Close(); err != nil {
+			t.Error(err)
+		}
+	}
+
+	readerConn, readerAddr := newLocalListener(t)
+	defer testCloser(readerConn)
+	writerConn := dialListener(t, readerAddr)
+	defer testCloser(writerConn)
+
+	var (
+		expected = Buffers{
+			[]byte("ABCDEFGHIJKLMNOPQRSTUVWXYZ"),
+			[]byte("0123456789"),
+			[]byte("abcdefghijklmnopqrstuvwxyz"),
+		}
+		received = func() (received Buffers) {
+			received = make(Buffers, len(expected))
+			for i := range received {
+				received[i] = make([]byte, len(expected[i]))
+			}
+			return
+		}()
+	)
+
+	writeMessages(t, writerConn, expected)
+	if _, err := received.ReadFrom(readerConn); err != nil {
+		t.Fatal(err)
+	}
+
+	for i, got := range received {
+		want := expected[i]
+		if !bytes.Equal(got, want) {
+			t.Errorf("buffer %d did not match expected data"+
+				"\n\tgot: %s"+
+				"\n\twant: %s",
+				i, got, want,
+			)
+		}
+	}
+}
+
+func newLocalListener(t *testing.T) (*net.UDPConn, *net.UDPAddr) {
```

Comment:
1) UDP was chosen here mainly for compat and simplicity, but could easily be changed to use Unix domain sockets instead.
I could be wrong, but I don't think the message passing semantics differ between them (especially when local).
In addition, (modern) Unices and Windows typically support UDS, so instead of letting the system choose a UDP port for us, we could create (and cleanup) socket files during the test. But I don't think we have to make this change, so I won't unless requested.


2) It should be noted that pkg `net` (as of 1.19) implies all the `MsgX` methods are not implemented on Plan9 regardless of transport.